### PR TITLE
Fix key counter counting in break time

### PR DESCRIPTION
**Pull request by Terochi**
Fixes https://github.com/ppy/osu/discussions/23272

### CODE REVIEW

**Comment by peppy** [comment]
See failing tests.

**Comment by Terochi** [comment]
I don't think those are related.

File: osu.Game/Screens/Play/HUD/DefaultKeyCounterDisplay.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ protected override void Update()
                 FadeTime = key_fade_time,
                 KeyDownTextColor = KeyDownTextColor,
                 KeyUpTextColor = KeyUpTextColor,
+                IsCounting = { BindTarget = IsCounting }
```
Comment (bdach) [inline osu.Game/Screens/Play/HUD/DefaultKeyCounterDisplay.cs:45]: I'm not a huge fan of the implementations having to hook this up themselves. Ideally it should be done automatically by the base class.

Else we're 99% going to hit this same rake with every new counter implementation.

**Comment by Terochi** [comment]
My only ideas would be something like
`IsCounting.BindValueChanged(change => Counters.ForEach(co => co.IsCounting.Value = change.NewValue));`
(maybe even make the `KeyCounter`s `IsCounting` a normal bool, as the bindable is not used anywhere yet)
or
```cs
// KeyCounterDisplay.cs
protected abstract KeyCounter GetCounter(InputTrigger trigger);
public void Add(InputTrigger trigger) => GetCounter(trigger).IsCounting.BindTo(IsCounting);
```

**Comment by peppy** [comment]
The test failures are 100% related. Please run tests locally.

**Comment by Terochi** [comment]
Sorry, this is a big mistake from me. I had a working implementation, which I scratched, and when I tried to implement it again, I forgot to remove the previous binding. Should probably learn to take my time.

File: osu.Game/Screens/Play/HUD/DefaultKeyCounterDisplay.cs [hunk 2]
```diff
@@ -36,13 +36,19 @@ protected override void Update()
...
+            var keyCounter = new DefaultKeyCounter(trigger)
             {
                 FadeTime = key_fade_time,
                 KeyDownTextColor = KeyDownTextColor,
-                KeyUpTextColor = KeyUpTextColor,
-            });
+                KeyUpTextColor = KeyUpTextColor
+            };
+
+            keyFlow.Add(keyCounter);
```
Comment (peppy) [inline osu.Game/Screens/Play/HUD/DefaultKeyCounterDisplay.cs:48]: This should not be in a `Create` method.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Play/HUD/DefaultKeyCounterDisplay.cs (hunk 2 above)
Comment (bdach) [reply osu.Game/Screens/Play/HUD/DefaultKeyCounterDisplay.cs]: I mean I agree, but a few renames (`Create` -> `AddInternal`) would have made this passable I think...

**Comment by bdach** [comment]
> Also has the option to disable each `KeyCounter` separately.

Why would you want that though.

**Pull request closed without merging**